lib/seattleflu/id3c/cli/command/clinical.py

```diff
@@ -1071,117 +1084,168 @@ def parse_kp2023(kp2023_filename: str) -> None:
     dump_ndjson(clinical_records)
 
 
-def map_icd10_codes(df: pd.DataFrame) -> pd.DataFrame:
+def convert_column_set_to_binary(df: pd.DataFrame, prefix: str) -> pd.DataFrame:
+    """
+    Given a DataFrame *df* of clinical records and a string *prefix* with
+    a prefix denoting which columns to convert, returns a DataFrame where
+    columns whose names begin with *prefix* contain only values 0, 1, or None.
+    Any value other than 1 in the input column is converted to None.
+    See KP2023 data dictionary for details.
+    """
+    cols = [c for c in df.columns if c.startswith(prefix)]
+    for c in cols:
+        df.loc[df[c] > 1, c] = None
+
+    return df
+
+
+def column_set_is_binary(df: pd.DataFrame, prefix:str) -> bool:
+    """
+    Given a DataFrame *df* of clinical records and a string *prefix* with
+    a prefix denoting columns of interest, returns True if all columns
+    beginning with the provided prefix contain only 0/1/None values,
+    otherwise returns False.
+    """
+    cols = [c for c in df.columns if c.startswith(prefix)]
+    for c in cols:
+        if not np.isin(df[c].dropna().unique(), [0, 1]).all():
+            return False
+    
+    return True
```

Comment:
Could shorten this block to something like:
```suggestion
    return all([np.isin(df[c].dropna().unique(), [0, 1]).all() for c in cols])
```
but not a big deal (and is probably less readable).